python/ccf/ledger.py

```diff
@@ -496,24 +496,29 @@ def _reset_iterators(self):
         # Initialize LedgerValidator instance which will be passed to LedgerChunks.
         self._ledger_validator = LedgerValidator()
 
-    def __init__(self, directory: str):
+    def __init__(self, directories: List[str]):
 
         self._filenames = []
 
-        ledgers = os.listdir(directory)
+        ledger_files = []
+        for directory in directories:
+            for path in os.listdir(directory):
+                chunk = os.path.join(directory, path)
```

Comment:
Should we be defensive here and check that `path` hasn't already been read, e.g. if the same directory is passed twice, or if the same ledger chunk (based on its name only) exists in two directories?

Reply:
If the same chunk occurs twice, there will be a validation error in the signatures. Properly making sure there are no duplicates (ie. resolving symlinks etc) may be more trouble than it's worth in practice.

Reply:
Fair point. I thought we could easily prevent suspicious signature verification errors here, but resolving symlinks, etc. seems too convoluted indeed.

Reply:
I suspect we can use the parsing of the sequence numbers to implement an easy but still helpful check.